math/modulararithmetic/modularexponentiation_test.go

```diff
@@ -0,0 +1,57 @@
+package modulararithmetic
+
+import "testing"
+
+type cases struct {
+	name        string
+	description string
+	base        int64
+	exponent    int64
+	mod         int64
+	expected    int64
+}
+
+var testCases = []cases{
```

Comment:
Should tests include negative numbers to prove that we pass and fail properly?

Reply:
That's a good shout, I think I should add that

Reply:
Although, now that I think about it, I don't know how to do that. Any ideas?